### PR TITLE
chore(lint): turn on the 'no-var' rule

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
adding because it is turned on in the `@typescript-eslint` base eslint rule configuration file, here:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/eslint-recommended.ts#L27

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

### CODE REVIEW

File: src/client/polyfills/css-shim/test/css-shim.spec.ts [hunk 1]
```diff
@@ -366,8 +366,8 @@ describe('css-shim', () => {
     );
   });
 
-  var window: Window;
```
Comment (rwaskiewicz) [inline src/client/polyfills/css-shim/test/css-shim.spec.ts:369]: I think for our shims, it's safer to leave `var` in place. They (the shims) are designed to run in older browsers (like Safari 10, IE 11). But I think these shims will go away in a future major release where we don't support those browsers. For now, can we ignore 'no-var' here?

File: src/dev-server/dev-server-client/polyfills.ts [hunk 2]
```diff
@@ -24,12 +24,12 @@ const applyObjectAssign = () => {
   if (typeof Object.assign !== 'function') {
```
Comment (rwaskiewicz) [inline src/dev-server/dev-server-client/polyfills.ts:24]: I think for our shims, it's safer to leave `var` in place. They (the shims) are designed to run in older browsers (like Safari 10, IE 11). But I think these shims will go away in a future major release where we don't support those browsers. For now, can we ignore 'no-var' here?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: src/client/polyfills/css-shim/test/css-shim.spec.ts (hunk 1 above)
Comment (alicewriteswrongs) [reply src/client/polyfills/css-shim/test/css-shim.spec.ts:369]: sure, that makes sense, I'll make that edit now

**Comment by alicewriteswrongs** [comment]
@rwaskiewicz added ignores for those `var` usages, and then also dealt with rebasing the branch over `main`

File: src/client/polyfills/css-shim/test/css-shim.spec.ts [hunk 3]
```diff
@@ -366,8 +366,8 @@ describe('css-shim', () => {
...
-  var document: Document;
+  var window: Window; // eslint-disable-line no-var
+  var document: Document; // eslint-disable-line no-var
```
Comment (rwaskiewicz) [inline src/client/polyfills/css-shim/test/css-shim.spec.ts:370]: One last thing - can we add the justification for disabling eslint here?
```suggestion
  var window: Window; // eslint-disable-line no-var -- shims will continue to use var while we support older browsers
  var document: Document; // eslint-disable-line no-var -- shims will continue to use var while we support older browsers
```

File: src/dev-server/dev-server-client/polyfills.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-var */
```
Comment (rwaskiewicz) [inline src/dev-server/dev-server-client/polyfills.ts:1]: One last thing - can we add the justification for disabling eslint here?
```suggestion
/* eslint-disable no-var -- shims will continue to use var while we support older browsers */
```

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

**Comment by alicewriteswrongs** [comment]
@rwaskiewicz just added eslint disable justification strings

**Pull request merged**